gkr/src/prover/linear_gkr.rs

```diff
@@ -168,6 +168,7 @@ impl<Cfg: GKRConfig> Prover<Cfg> {
         pcs_scratch: &mut <Cfg::PCS as PCSForExpanderGKR<Cfg::FieldConfig, Cfg::Transcript>>::ScratchPad,
         transcript: &mut Cfg::Transcript,
     ) {
+        transcript.lock_proof();
```

Comment:
why do you need to lock proof (seems irrelevant to this PR)

Reply:
I don't recall the exact answer, but I do remember the context - supposing the PCS is instantiated from some multi-round IOP, WLOG we say FRI as an example to carry on - then the PCS-IOP challenge should be irrelevant of the whole GKR proof, but rather inferred from the Fiat-Shamir transcript.  In this way, the PCS internal IOP challenge should not get into the transcript proof, and thus we should add a context-alike thing to lock and unlock the transcript proof, just in case some day we put FRI into the codebase (though we are unsure about that).